### PR TITLE
=rem Improve the TimeoutReason error message

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

**Comment by patriknw** [comment]
@drewhk please review, anything else that should be included in the err message?

File: akka-remote/src/main/scala/akka/remote/transport/AkkaProtocolTransport.scala [hunk 1]
```diff
@@ -463,7 +463,8 @@ private[transport] class ProtocolStateActor(initialData: InitialProtocolStateDat
     } else {
       // send disassociate just to be sure
       sendDisassociate(wrappedHandle, Unknown)
-      stop(FSM.Failure(TimeoutReason("No response from remote. Handshake timed out or transport failure detector triggered.")))
+      stop(FSM.Failure(TimeoutReason(s"No response from remote when $stateName. " +
```
Comment (drewhk) [inline akka-remote/src/main/scala/akka/remote/transport/AkkaProtocolTransport.scala]: Does the stateName really help users? Maybe put it at the end of the sentence in a parenthesis:

    (internal state was $stateName)

To indicate it is DEBUG info for us mostly

**Comment by drewhk** [comment]
LGTM apart from the small nitpick

File: akka-remote/src/main/scala/akka/remote/transport/AkkaProtocolTransport.scala (hunk 1 above)
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/transport/AkkaProtocolTransport.scala]: good point

**Comment by ktoso** [comment]
LGTM

**Pull request merged**